File: algorithms/python/quick_sort.py
def quicksort(values):
    if len(values) <= 1:
        return values

    less_than_pivot = []
    greater_than_pivot = []
    pivot = values[0]

    for value in values[1:]:
        if value <= pivot:
            less_than_pivot.append(value)
        else:
            greater_than_pivot.append(value)

    return quicksort(less_than_pivot) + [pivot] + quicksort(greater_than_pivot)


numbers = [8, 3, 19, 43, 17, 22, 1, 4, 15, 2, 16, 32, 5]
sorted_numbers = quicksort(numbers)
print(sorted_numbers)

File: algorithms/go/quicksort.go
package main

import "crypto/rand"

func generateSlice(size int) []int {
	arr := make([]int, size)
	for i := 0; i < size; i++ {
		arr[i] = i
	}
	return arr
}

func quicksort(arr []int) []int {
	if len(arr) < 2 {
		return arr
	}

	left, right := 0, len(a)-1

	pivot := rand.Int() % len(a)

	arr[pivot], arr[right] = arr[right], arr[pivot]

	for a, v := range arr {
		if v < arr[right] {
			arr[left], arr[a] = arr[a], arr[left]
			left++
		}
	}

	arr[left], arr[right] = arr[right], arr[left]

	quicksort(arr[:left])
	quicksort(arr[left+1:])

	return arr
}
